xbmc/cores/VideoPlayer/Interfaces/IVPClockCallback.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+/*
+ *      Copyright (C) 2015 Team Kodi
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+class CDVDClock;
+
+class IVPClockCallback
+{
+public:
+  virtual CDVDClock* GetClock() = 0;
```

Comment:
It is more robust if we don't pass the entire clock object. This interface should implement methods like GetClock. This way I did it for AE: https://github.com/xbmc/xbmc/blob/master/xbmc/cores/AudioEngine/Interfaces/AEStream.h#L34